### PR TITLE
Traffic split auto redirection should only consider the active revisions 

### LINKED ISSUE #1609: Traffic split calculation should only use traffic-split revisions
When creating a traffic split and automatically calculating a percentage rest, all revisions are taken into account, even those the are currently nor part of the traffic distribution:

```
$ kn revision list
NAME           SERVICE   TRAFFIC   TAGS   GENERATION   AGE   CONDITIONS   READY   REASON
random-00002   random    100%             2            37s   3 OK / 4     True
random-00001   random                     1            66s   4 OK / 4     True

$ kn service update random --image rhuss/random:2.0 --traffic 50
Error: unable to allocate the remaining traffic: 50%. Reason: incorrect number of revisions specified. Only 1 revision should be missing
Run 'kn --help' for usage
```

Here only `random-00002` should be taken into account and automatically reduced to 50% (so there should be no error message, but the expected end result should be `random-0002` set to 50% and the new revision `random-00003` should be 50%.

### PR DESCRIPTION
**Pull request by vyasgun**
## Changes

* Automatic redirection of traffic should only take the active revisions into consideration
* Modified unit test according to the change

```
gvyas-mac:client gvyas$ ./kn revision list -s cloudevents-player
NAME                       SERVICE              TRAFFIC   TAGS   GENERATION   AGE   CONDITIONS   READY   REASON
cloudevents-player-00005   cloudevents-player   100%             5            76m   3 OK / 4     True    
cloudevents-player-00004   cloudevents-player                    4            84m   3 OK / 4     True    
cloudevents-player-00003   cloudevents-player                    3            85m   3 OK / 4     True    
cloudevents-player-00002   cloudevents-player                    2            27d   3 OK / 4     True    
cloudevents-player-00001   cloudevents-player                    1            27d   3 OK / 4     True    
```
```
gvyas-mac:client gvyas$ ./kn service update cloudevents-player --env ABC=xyz --traffic 70
Updating Service 'cloudevents-player' in namespace 'default':

  0.065s The Route is still working to reflect the latest desired specification.
  0.110s ...
  3.198s Traffic is not yet migrated to the latest revision.
  3.262s Ingress has not yet been reconciled.
  3.309s Waiting for load balancer to be ready
  3.458s Ready to serve.

Service 'cloudevents-player' updated to latest revision 'cloudevents-player-00006' is available at URL:
http://cloudevents-player.default.10.111.180.117.sslip.io
gvyas-mac:client gvyas$ ./kn revision list -s cloudevents-player
NAME                       SERVICE              TRAFFIC   TAGS   GENERATION   AGE   CONDITIONS   READY   REASON
cloudevents-player-00006   cloudevents-player   70%              6            6s    4 OK / 4     True    
cloudevents-player-00005   cloudevents-player   30%              5            77m   3 OK / 4     True    
cloudevents-player-00004   cloudevents-player                    4            86m   3 OK / 4     True    
cloudevents-player-00003   cloudevents-player                    3            86m   3 OK / 4     True    
cloudevents-player-00002   cloudevents-player                    2            27d   3 OK / 4     True    
cloudevents-player-00001   cloudevents-player                    1            27d   3 OK / 4     True    
```

## Reference

Fixes #1609

### CODE REVIEW

**Comment by dsimansk** [comment]
@vyasgun seems like an error related to the fix?
https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_client/1617/pull-knative-client-integration-tests-latest-release/1500785404868038656#1:build-log.txt%3A1162

File: pkg/kn/traffic/compute.go [hunk 1]
```diff
@@ -242,12 +242,12 @@ func errorSumGreaterThan100(sum int) error {
...
 // - if traffic as per flags < 100, can remaining traffic be automatically directed
-func verifyInput(trafficFlags *flags.Traffic, svc *servingv1.Service, revisions []servingv1.Revision, mutation bool) error {
+func verifyInput(trafficFlags *flags.Traffic, svc *servingv1.Service, targets []servingv1.TrafficTarget, existingRevisions []servingv1.Revision, mutation bool) error {
 	// check if traffic is being sent to @latest tag
 	var latestRefTraffic bool
 
-	// number of existing revisions
-	var existingRevisionCount = len(revisions)
+	// number of existing targets
+	var existingRevisionCount = len(targets)
```
Comment (rhuss) [inline pkg/kn/traffic/compute.go:250]: would it make sense to rename the variable to `targetsCount` ?

**Review by rhuss — COMMENTED** [review]
Looks good to me (just one minor cosmetic comment)

**Comment by rhuss** [comment]
Also, tested it locally, and works as expected. Thanks @vyasgun !

**Comment by vyasgun** [comment]
Taking a look at the failing tests

File: pkg/kn/traffic/compute.go (hunk 1 above)
Comment (vyasgun) [reply pkg/kn/traffic/compute.go:250]: Yes, I'll change it

**Pull request merged**